tools/contractdeployer/contract_deployer.go

```diff
@@ -0,0 +1,104 @@
+package contractdeployer
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/obscuronet/obscuro-playground/go/common/log"
+	"github.com/obscuronet/obscuro-playground/go/ethadapter"
+	"github.com/obscuronet/obscuro-playground/go/ethadapter/mgmtcontractlib"
+	"github.com/obscuronet/obscuro-playground/go/wallet"
+	"github.com/obscuronet/obscuro-playground/integration/erc20contract"
+)
+
+type ContractDeployer struct {
+	config *Config
+}
+
+func NewContractDeployer(config *Config) *ContractDeployer {
+	return &ContractDeployer{
+		config: config,
+	}
+}
+
+func (cd *ContractDeployer) Run() error {
+	// connect to the L1
+	privateKey, err := crypto.HexToECDSA(cd.config.PrivateKey)
+	if err != nil {
+		return fmt.Errorf("could not recover private key from hex. Cause: %w", err)
+	}
+
+	// load the wallet
+	w := wallet.NewInMemoryWalletFromPK(cd.config.ChainID, privateKey)
+	// connect to the l1
+	client, err := ethadapter.NewEthClient(cd.config.L1NodeHost, cd.config.L1NodePort, 30*time.Second)
+	if err != nil {
+		return fmt.Errorf("unable to connect to the l1 host: %w", err)
+	}
+
+	// deploy the contracts
+	contractAddr, err := deployContract(client, w, common.Hex2Bytes(mgmtcontractlib.MgmtContractByteCode))
+	if err != nil {
+		return fmt.Errorf("unable to deploy contract to the l1 host: %w", err)
+	}
+	erc20contractAddr, err := deployContract(client, w, common.Hex2Bytes(erc20contract.ContractByteCode))
+	if err != nil {
+		return fmt.Errorf("unable to deploy contract to the l1 host: %w", err)
+	}
+
+	// print the contract address
+	fmt.Printf("{\"MgmtContractAddr\":\"%s\", \"ERC20ContractAddr\":\"%s\"}",
+		contractAddr.Hex(),
+		erc20contractAddr.Hex(),
+	)
+	time.Sleep(5 * time.Second)
```

Comment:
@otherview I think it's unintuitive to sleep after the last print, will look like it's hung. I'd suggest either sleeping before printing the comntract addresses, or printing some message before the sleep explaining why it's there. Alternatively, can we check using the L1 client where the deploy has completed?

Reply:
Done 👍